meetings/2022-03-30.md

```diff
@@ -0,0 +1,89 @@
+---
+title: 2022-03-30
+parent: Meetings
+---
+# ASWF CI Working Group
+
+Meeting:   30 March 2022
+
+[https://zoom.us/j/757849640?pwd=QzE1K2hrL2FHSFhKK3h5Z3BWTFJsZz09](https://zoom.us/j/757849640?pwd=QzE1K2hrL2FHSFhKK3h5Z3BWTFJsZz09)
+
+## Attendees
+
+* Jean-Francois Panisset (VES Technology Committee)
+* Brent Villalobos (Dreamworks)
+* Andrew Grimberg (LF RelEng)
+* Ryan Botriell (spk / ILM)
+* Aloys Baillet (NVIDIA)
+* Jean-Christophe Morin (Signiant)
+* Rob Fanner (Foundry)
+* James Pedlingham (Foundry)
+* Tom Saunders (FuseFX)
+
+## Apologies
+
+* Christina Tempelaar-Lietz (Epic Games)
+* Larry Gritz (Sony Imageworks)
+
+## New items
+
+* Conan at Foundry (Rob Fanner et al)
+    * Production proven Conan packages used for Foundry builds with namespacing, multi-platform support...
+    * Discussion on general usefulness / contribution
+    * Robert is a manager, James is the maintainer of the packages.
+    * Started using Conan to help with VFX Reference Platform 2017, Nuke first, then Mari and Katana, and also in Modo.
+    * Adds namespace support, USD / OCIO color spaces, symbol hiding, compiler variants.
+    * Nuke hides Boost symbols through symbol hiding, Katana needs to provide Python bindings for USD.
+    * Sample Conan recipes may not be used for production, so had to do their own.
+    * Lots of README documentation for developer workflows, how to update packages, how to patch the recipes, the libraries.
+    * Add SPDX license info
+    * Introduced config.yml files for each recipe to indicate which variants are used for recipes, all the ways a package can be built (static/dynamic, debug/release...). Config.yml says which versions are used.
+    * Rudimentary CI system, can see how builds progress through GitLib.
+    * Helped deal with complexity to follow VFX Platform more quickly, help with Apple Silicon. Each product may have multiple versions of USD, up to 64 build variants, Conan helps support this.
+    * Biggest win was to use expert knowledge and encoding it in a scripted format. Has introduced more overhead for regular developers who know how to work with Conan caches (can get out of sync). Recipes help with upgrades.
+    * Some gotchas, trying to put more guardrails around build system. There are gotchas, "can't get from here to there", for instance may know you don't need to rebuild libz, but Conan is stricter. Looking at ways to override some of these.
+    * Trying to express known compatibilities in the compiler toolchain.
+    * Had thought about publishing as open source, but who would use it? Industry likely uses Rez? Also interest in vcpkg. Also use the aswf-docker containers, helps with quick and easy getting started work, but production is Conan.
+    * Because didn't open source, has grown some "proprietary warts" that would need to be cleaned out before cleaned out.
```

Comment:
The second "cleaned out" should probably be something like "releasing it"?

Reply:
Fixed

Reply:
Thank you for proof reading!